### DISCUSSION TITLE
*page.client.vue returns empty html content and the recommended ClientOnly Component does not stop the component from executing on the server side

### DISCUSSION THREAD

**Discussion by apappas1129** [vikejs/vike#1036] [Discussion]
I find it hard at the moment to recreate a minimal project with the reproducible issue due to other headaches in my current R&D. But you can checkout [this repo](https://github.com/apappas1129/vue-training) and [run it](https://github.com/apappas1129/vue-training#running-the-project) locally.

The first concern I have here is that I don't exactly know how to implement a client-only-rendered Vue page component.
I thought I understood how to do it since [this one](https://github.com/apappas1129/vue-training/blob/main/src/pages/subjects/create/index.page.client.vue)worked perfectly fine and prevented the vue3-tabs components from running on the server side. But then when I try to reimplement it with a `*page.server.ts` alongside it for prefetching data in `onBeforeRender`, apparently the page is blank without any errors other than the warning:
```
The 3rd HTML variable is `undefined` which will be converted to an empty string, see render() hook defined by /src/renderer/_default.page.server.ts > `export { render }`. Pass an empty string instead of `undefined` to remove this warning.
```
Which surprisingly is hard for me to find where to fix and suppress this warning.

I would appreciate your help to make just a simple example of what I need that I can simply replicate then to wrap up my training project.

**Comment by apappas1129** [comment]
For the author, or anyone who may offer a little of their time to chat and share-screen with me, I am looking into giving [a small one-time sponsorship](https://github.com/sponsors/brillout?frequency=one-time) to kinda pay even for at least 15mins of your time. Desperately need help here to straighten things up in my ssr related code.

**Reply by brillout** [reply]
The thing is that most of your issues seem to live on the user land and to be unrelated to VPS itself. Note that the upcoming `vike-vue` integration will provide you with an easy get started akin to Nuxt.

**Reply by apappas1129** [reply]
Yes it really is on my implementation details which I desperately need help on. I really want to figure this thing out. The point here is I want to experience all the shenanigans (at least the basics) happening on SSR side of things before I dive in to Nuxt or whatever that abstracts almost all of this things away and spoils the user. And I find your VPS the perfect tool for my training project.

**Reply by brillout** [reply]
Figuring it out on your own *is* the learing experience.
